webv2/helpers/helpers.go

```diff
@@ -76,31 +76,48 @@ func createDatabase(ctx context.Context, uri string) error {
 				SchemaConversionObject JSON NOT NULL,
 				CreateTimestamp TIMESTAMP NOT NULL,
 			  ) PRIMARY KEY(VersionId)`,
-			`CREATE TABLE SMT_JOBS (
+			`CREATE TABLE SMT_JOB (
 				JobId STRING(100) NOT NULL,
 				JobName STRING(100) NOT NULL,
 				JobType STRING(100) NOT NULL,
+				JobStateData JSON,
 				JobData JSON,
 				Dialect STRING(50) NOT NULL,
 				SpannerDatabaseName STRING(100) NOT NULL,
-				CreatedAt TIMESTAMP NOT NULL,
-			  ) PRIMARY KEY(JobId)`,
-			`CREATE TABLE SMT_RESOURCES (
+				UpdatedAt TIMESTAMP NOT NULL OPTIONS (allow_commit_timestamp=true),
+		  	) PRIMARY KEY(JobId)`,
+			`CREATE TABLE SMT_JOB_HISTORY (
+				JobId STRING(100) NOT NULL,
+				Version INT64 NOT NULL,
+				JobName STRING(100) NOT NULL,
+				JobType STRING(100) NOT NULL,
+				JobStateData JSON,
+				JobData JSON,
+				Dialect STRING(50) NOT NULL,
+				SpannerDatabaseName STRING(100) NOT NULL,
+				CreatedAt TIMESTAMP NOT NULL OPTIONS (allow_commit_timestamp=true),
+		  	) PRIMARY KEY(JobId, Version)`,
+			`CREATE TABLE SMT_RESOURCE (
 				ResourceId STRING(100) NOT NULL,
 				JobId STRING(100) NOT NULL,
-				ExternalId STRING(100) NOT NULL,
+				ExternalId STRING(100),
 				ResourceName STRING(100) NOT NULL,
 				ResourceType STRING(100) NOT NULL,
+				ResourceStateData JSON,
 				ResourceData JSON,
-				CreatedAt TIMESTAMP NOT NULL,
-			  ) PRIMARY KEY(ResourceId)`,
-			`CREATE TABLE SMT_STATES (
-				StateId STRING(100) NOT NULL,
-				StateVersion INT64 NOT NULL,
+				UpdatedAt TIMESTAMP NOT NULL OPTIONS (allow_commit_timestamp=true),
```

Comment:
This should have both `createdAt` and `updatedAt` right?

Reply:
Based on the discussion, we decided UpdatedAt will act as the "CreatedAt" value at first, but subsequent updates will overwrite it. The history table would would be used if we need the first created at timestamp.

Reply:
I think I meant that CreatedAt and UpdatedAt were both part of the live table. I understand we can find the earliest timestamp using the history but storing both of these together is typically a pretty standard optimization/schema design.

Reply:
Done